### ISSUE TITLE
separate lib from middleware

### ISSUE THREAD

**Issue by jonathanong** [defunctzombie/polyfill-middleware#1]
would be cool if the polyfill-creating portion was separated into its own lib, then have the middleware be a separate lib again. just because.

**Comment by defunctzombie** [comment]
I agree. I kinda think the polyfill creating portion could exist in the base `polyfill` lib. Tho there are actually some opinions with that because you could technically write something that uses streams during polyfill source bundling and sends that to the end user.

So for now I went with just having it here as a starting point. I tried to separate out logically to make that transition easier later.

**Comment by jonathanong** [comment]
well that's just an optimization. i doubt it'll be worth using streams though since this is pretty small and you could just store it in memory.

 +1 to including it in the lib though - was going to add that but what you wrote sounded like the opposite so i was like nvm... haha
